sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -93,6 +93,14 @@ def _is_cloud_event(event):
     except TypeError:
         return False
 
+def _is_eventgrid_event(event):
+    # type: (Any) -> bool
+    required = ('subject', 'event_type', 'data', 'data_version')
```

Comment:
dumb question: I see that in the `EventGridEvent` class, we have other required props:
```python
    _validation = {
        'id': {'required': True},
        'subject': {'required': True},
        'data': {'required': True},
        'event_type': {'required': True},
        'event_time': {'required': True},
        'metadata_version': {'readonly': True},
        'data_version': {'required': True},
    }
```

so we don't need id, event_time here?

Reply:
right - `id` is required, but when not provided, it's a random uuid4 ; similarly `event_time` is the current datetime.

Reply:
We document id and event_time as optional and have them as kwargs